### PR TITLE
GEOMESA-2690 Provide callback for handling kafka consumer errors

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala [hunk 1]
```diff
@@ -110,3 +114,21 @@ abstract class ThreadedConsumer(
...
+  trait ConsumerErrorHandler {
+
+    /**
+      * Invoked on a fatal error
+      *
+      * @param consumer consumer identifier
+      * @param e exception
+      * @return true to continue processing, false to terminate the consumer
+      */
+    def handle(consumer: String, e: Throwable): Boolean
```
Comment (jnh5y) [inline geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: Ok, this is cool.  How does one configure one?:)

Comment (elahrvivaz) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: you pass it off to the `startConsumers` method here: https://github.com/locationtech/geomesa/pull/2333/files#diff-7f9d3345259eca082a59f009c3f485a5R52

Comment (jnh5y) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: Ok, so one would have to do this quite manually?

Comment (elahrvivaz) [reply geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala]: Yeah, the idea is to give access to the extending class, as currently everything is private and inaccessible, but without exposing too much inner logic.

**Pull request merged**